### PR TITLE
Improve styling of sidebar entries that wrap

### PR DESCRIPTION
**Pull request by lorenzolewis**
#### What kind of changes does this PR include?

- Changes to Starlight code

#### Description

If a sidebar entry wraps to a new line then it can be difficult to distinguish that entry from the ones above/below it. This puts in a fix to visually indent any subsequent lines.

Before:
Screenshot 2023-07-27 at 14 17 49
Screenshot 2023-07-27 at 14 23 08

After:
Screenshot 2023-07-27 at 14 17 37
Screenshot 2023-07-27 at 14 23 17

### CODE REVIEW

**Review by delucis — CHANGES_REQUESTED** [review]
Thanks for this @lorenzolewis! I’ve noticed this issue before. I actually think the issue could be better addressed by a narrower line height? That would mean the wrapped lines nestle closer together and form a single visual unit more clearly. Been meaning to tackle that.

I think the requirement would be: 
- narrower line-height
- a little extra vertical padding so that the current appearance for single-line links is preserved.

**Comment by lorenzolewis** [comment]
Sure, I can get behind that. I'll play around with it a bit to dial it in and update

**Comment by lorenzolewis** [comment]
Screenshot 2023-07-28 at 21 52 49
Screenshot 2023-07-28 at 21 52 45

@delucis here's those changes, feel free to dial them as you see fit. I tried to keep the overall vertical height of an item as close as possible.

File: packages/starlight/components/SidebarSublist.astro [hunk 1]
```diff
@@ -87,7 +87,8 @@ interface Props {
 		border-radius: 0.25rem;
 		text-decoration: none;
 		color: var(--sl-color-gray-2);
-		padding: 0.1em var(--sl-sidebar-item-padding-inline);
+		padding: 0.3em var(--sl-sidebar-item-padding-inline);
+		line-height: 1.4em;
```
Comment (delucis) [inline packages/starlight/components/SidebarSublist.astro:91]: Nit but we generally use unit-less line-height values (has the same effect anyway):

```suggestion
		line-height: 1.4;
```

**Review by delucis — APPROVED** [review]
That was quick! Looks fantastic, thank you @lorenzolewis. One stylistic nit but that’s all I got.

**Pull request merged**